### PR TITLE
Fix/link display

### PR DESCRIPTION
**Pull request by SrinandhiniM**
Link feature in Post form is completed similar to Reddit App.

- Link has separate field in Link tab in Post form and it accepts only link.
- The link is displayed in Post List page and if it is clicked it redirects to the page in next tab.
- The link preview and link image is shown in post displaying page and it also redirects to the page in next tab.

### CODE REVIEW

File: app/views/posts/_post_display.html.erb [hunk 1]
```diff
@@ -1,27 +1,33 @@
-<div class = "post-body" style="word-wrap: break-word;">
-  <%= @post.body  %>
+<div class="post-body" style="word-wrap: break-word;">
+  <%= @post.body %>
+  <% if @post.link.to_s.match(URI.regexp) %>
```
Comment (iMagesh) [inline app/views/posts/_post_display.html.erb:3]: move this to helper

File: app/views/posts/_post_display.html.erb [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+            </div>
+          <% end %>
+          <a class="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
+            <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+            <span class="sr-only">Previous</span>
+          </a>
+          <a class="carousel-control-next" href="#carouselExampleControls" role="button" data-slide="next">
+            <span class="carousel-control-next-icon" aria-hidden="true"></span>
+            <span class="sr-only">Next</span>
+          </a>
```
Comment (iMagesh) [inline app/views/posts/_post_display.html.erb:28]: this can be moved to a separate partial

**Comment by SrinandhiniM** [comment]
Issues are resolved now.

- Helper method is created to check whether the post is link or not.
- Partial file is created for displaying the images in Post.

File: app/views/posts/_images_display.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<div id="carouselExampleControls" class="carousel slide" data-ride="carousel">
```
Comment (iMagesh) [inline app/views/posts/_images_display.html.erb:1]: can you change the file name to image_carousel or something like that? just to make it more relevant

**Comment by SrinandhiniM** [comment]
File renamed for partial file.

**Pull request merged**